extension/background/pageMetadata.js

```diff
@@ -10,7 +10,43 @@ export async function getSelection(tabId) {
 
 export async function getMetadata(tabId) {
   await content.lazyInject(tabId, "/background/pageMetadata-contentScript.js");
-  return browser.tabs.sendMessage(tabId, {
+  const metadata = await browser.tabs.sendMessage(tabId, {
     type: "getMetadata",
   });
+  metadata.canonical = cleanURL(metadata.canonical);
+  return metadata;
+}
+
+function cleanURL(string) {
+  const globalBlockedParams = [
+    "utm_source",
+    "utm_medium",
+    "utm_campaign",
+    "utm_term",
+    "utm_content",
+  ];
+
+  function getReducedParams(str) {
+    const url = new URL(str);
+    const params = url.searchParams;
+
+    globalBlockedParams.map(blockedParam => {
+      if (params.get(blockedParam)) {
+        params.delete(blockedParam);
+      }
+    });
+    return params.toString();
+  }
+
+  function buildURL(baseURL, params) {
+    if (Object.keys(params).length === 0) {
+      return baseURL;
+    }
+    const newURL = baseURL + "?" + params;
+    return newURL;
+  }
+
+  const baseURL = string.split("?")[0];
+  const params = getReducedParams(string);
```

Comment:
You can actually remove most of this. I'd rename `getReducedParams()` to `reduceParams()` and then have it return `url.toString()`. It will remove the parameters and rebuild the URL all on its own (`urlObject.toString()` rebuilds the URL)